src/index.js

```diff
@@ -36,5 +88,58 @@ const renderTodoList = () => {
     });
 };
 
+// Delete the specified task object from the "tasks" array, update the "index" property.
+deleteTodo = (index) => {
+  tasks = tasks.filter((task) => task.index !== index);
+  updateTodoIndexes();
+  saveTodos();
+  renderTodoList();
+};
+
+// Replace the description of the specified "task" object with an input element.
+editTodoDescription = (task) => {
+  const inputEl = document.createElement('input');
+  inputEl.type = 'text';
+  inputEl.value = task.description;
+  inputEl.classList.add('edit-input');
+
+  inputEl.addEventListener('keydown', (event) => {
+    if (event.key === 'Enter') {
+      task.description = inputEl.value.trim();
+      saveTodos();
+      renderTodoList();
+    } else if (event.key === 'Escape') {
+      renderTodoList();
+    }
+  });
+
+  const listItemEl = todoList.children[task.index];
+  listItemEl.replaceChild(inputEl, listItemEl.children[1]);
+  inputEl.select();
+};
+
+// Create a new task object with the specified "description", add it to the "tasks" array.
+function addNewTodo(description) {
+  const taskIndex = tasks.length;
+
+  const task = { description, completed: false, index: taskIndex };
+  tasks.push(task);
+  saveTodos();
+
```

Comment:
- [ ] Please make sure the index of the elements starts from 1 and not zero. This is a project requirement. the trick is in line 123
`  const task index = tasks.length + 1` 😄 
![index](https://github.com/Megagig/To-do-list/assets/99656687/260c1805-bcab-46b0-8f8e-3571fa00386a)